apps/src/code-studio/progress.js

```diff
@@ -289,33 +259,8 @@ function queryUserProgress(store, scriptData, currentLevelId) {
         pageType
       );
     }
-
-    if (data.focusAreaStageIds) {
-      store.dispatch(
-        updateFocusArea(data.changeFocusAreaPath, data.focusAreaStageIds)
-      );
-    }
-
-    if (data.lockableAuthorized) {
-      store.dispatch(authorizeLockable());
-    }
-
-    if (data.completed) {
-      store.dispatch(setScriptCompleted());
-    }
-
-    // Merge progress from server (loaded via AJAX)
-    if (data.levels) {
-      const levelProgress = _.mapValues(data.levels, getLevelResult);
-      store.dispatch(mergeProgress(levelProgress));
-      if (data.peerReviewsPerformed) {
-        store.dispatch(mergePeerReviewProgress(data.peerReviewsPerformed));
-      }
-      if (data.current_stage) {
-        store.dispatch(setCurrentStageId(data.current_stage));
-      }
-    }
-  });
+  };
+  store.dispatch(reduxQueryUserProgress(userId, onComplete));
```

Comment:
you don't need to pass a callback here, rather you can chain this like: `store.dispatch((reduxQueryUserProgress(userId)).then(onComplete)`. and then, depending on how the code is looking and your personal preference, you could inline `onComplete`.

This is possible because we are using `redux-thunk`. the docs are here: https://github.com/reduxjs/redux-thunk#composition

Here is an example in our own code: https://github.com/code-dot-org/code-dot-org/blob/502e34ec385179cb49237946fc5ee8f0095a45a9/apps/src/sites/studio/pages/teacher_dashboard/show.js#L81

You don't have to do it this way, but it's important to know about because sometimes a callback can throw a wrench into the gears of an otherwise manageable promise chain.

Reply:
🎉thank you, this simplifies the code _a lot_!